source/exe/hot_restart.h

```diff
@@ -138,8 +138,12 @@ class HotRestartImpl : public HotRestart,
   };
 
   struct RpcBase {
-    RpcBase(RpcMessageType type, uint64_t length = sizeof(RpcBase))
-        : type_(type), length_(length) {}
+    RpcBase(RpcMessageType type, uint64_t length = sizeof(RpcBase)) {
+      static_assert(std::is_standard_layout<RpcBase>::value, "is_standard_layout<RpcBase>");
```

Comment:
This was a good catch, since the subclasses aren't standard layout it turns out. Using SO as a shortcut to the C++11 definitions: https://stackoverflow.com/questions/7160901/why-is-c11s-pod-standard-layout-definition-the-way-it-is

It looks like since we have non-static member variables in both base and derived class, we're technically not standard layout.

We could avoid the inheritance at the expense of a little boiler plate.